app/controllers/YourAccountController.scala

```diff
@@ -50,43 +53,56 @@ class YourAccountController @Inject()(
     implicit request =>
       val results = getPeriodsAndFinancialData()
 
-      results.map {
+      results.flatMap {
         case (Right(availablePeriodsWithStatus), Right(vatReturnsWithFinancialData)) =>
           val filteredPeriodsWithOutstandingAmounts =
             getFilteredPeriodsWithOutstandingAmounts(vatReturnsWithFinancialData)
           val duePeriodsWithOutstandingAmounts =
             filteredPeriodsWithOutstandingAmounts.filterNot(_.vatReturn.period.isOverdue(clock))
           val overduePeriodsWithOutstandingAmounts =
             filteredPeriodsWithOutstandingAmounts.filter(_.vatReturn.period.isOverdue(clock))
+          val sortedPeriodsToSubmit = availablePeriodsWithStatus
+            .filter(p => p.status == SubmissionStatus.Due || p.status == SubmissionStatus.Overdue).
+            sortBy(_.period.firstDay.toEpochDay)
 
-          Ok(view(
-            request.registration.registeredCompanyName,
-            request.vrn.vrn,
-            availablePeriodsWithStatus
-              .filter(_.status == SubmissionStatus.Overdue)
-              .map(_.period),
-            availablePeriodsWithStatus
-              .find(_.status == SubmissionStatus.Due)
-              .map(_.period),
-            duePeriodsWithOutstandingAmounts,
-            overduePeriodsWithOutstandingAmounts,
-            filteredPeriodsWithOutstandingAmounts.exists(_.charge.isEmpty)
-          ))
+             sortedPeriodsToSubmit.headOption.map(p => sessionRepository.get(request.userId, p.period)).getOrElse(Future.successful(None))
```

Comment:
yes - because we can only have one return in progress at any given time and it always has to be the oldest one, so we get all the returns available to the user, sort them and then get the first one as this one will always be the oldest, and we check if it is in progress or not